File: src/creational/prototype/Prototype.cs

using System;

// Prototype interface
public interface IPrototype
{
    IPrototype Clone();
}

// Concrete Prototype 1
public class ConcretePrototype1 : IPrototype
{
    public IPrototype Clone()
    {
        Console.WriteLine("Cloning ConcretePrototype1.");
        return (IPrototype)this.MemberwiseClone();
    }
}

// Concrete Prototype 2
public class ConcretePrototype2 : IPrototype
{
    public IPrototype Clone()
    {
        Console.WriteLine("Cloning ConcretePrototype2.");
        return (IPrototype)this.MemberwiseClone();
    }
}

// Client code
class Program
{
    static void Main()
    {
        IPrototype prototype1 = new ConcretePrototype1();
        IPrototype clone1 = prototype1.Clone();

        IPrototype prototype2 = new ConcretePrototype2();
        IPrototype clone2 = prototype2.Clone();
    }
}

File: creational/prototype/prototype.py

import copy

# Prototype interface
class Prototype:
    def clone(self):
        pass

# Concrete Prototype 1
class ConcretePrototype1(Prototype):
    def clone(self):
        print("Cloning ConcretePrototype1.")
        return copy.deepcopy(self)

# Concrete Prototype 2
class ConcretePrototype2(Prototype):
    def clone(self):
        print("Cloning ConcretePrototype2.")
        return copy.deepcopy(self)

if __name__ == "__main__":
    prototype1 = ConcretePrototype1()
    clone1 = prototype1.clone()

    prototype2 = ConcretePrototype2()
    clone2 = prototype2.clone()
